extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java

```diff
@@ -111,7 +110,7 @@ void build(
                                 DataSources.TRACING_DRIVER_CLASSNAME));
             }
 
-            if (aggregatedDataSourceBuildTimeConfig.getJdbcConfig().telemetry && !otelJdbcInstrumentationActive) {
+            if (aggregatedDataSourceBuildTimeConfig.getJdbcConfig().telemetry) {
```

Comment:
I simplified the `if` here as the extra condition was making it a bit more obscure than truly necessary.

Reply:
My thinking was that boolean evaluation is quicker than variable assignment :-) Simplified `if` is fine.

Reply:
So yes in theory. It might be of importance in a very hot path but it's not one and it's probably better to keep the code more readable.
But as I said, this one is debatable. Your pattern is more efficient but my impression is that it isn't worth reducing the readability.